docs/design/operator.md

```diff
@@ -0,0 +1,414 @@
+configuration, and management of these drivers using Custom Resource
+Definitions (CRDs).
+
+## System Overview
+
+The Ceph-CSI Operator is a Kubernetes operator that simplifies the management
+of Ceph CSI drivers. It provides a set of CRDs to manage the configurations of
+CephFS, RBD, and NFS drivers, as well as connection details to Ceph clusters.
+The operator ensures that the desired state specified in the CRDs is maintained
+in the cluster.
+
+## Assumptions and Dependencies
+
+Kubernetes cluster is available and operational. Ceph cluster is set up and
+accessible from the Kubernetes cluster. Users are familiar with Kubernetes and
+Ceph concepts. CRD and operator versions are compatible with the Kubernetes
+version in use.
+
+## General Constraints
+
+Resource limitations of the Kubernetes cluster (CPU, memory). Network latency
+and connectivity between the Kubernetes cluster and Ceph cluster. Watching for
+changes on the ceph cluster and updating internal details.  Security policies
+and RBAC configurations in the Kubernetes cluster.
+
+## Goals and Guidelines
+
+Automate the installation and management of Ceph-CSI drivers. Provide a
+consistent and flexible configuration mechanism using CRDs. Enable
+namespace-scoped configurations for different Ceph-CSI drivers. Ensure high
+availability and scalability of the operator. When running operator in a
+namespace provide an option to watch for the CR's in the  namespace where its
+running for ceph-CSI management. User will have an option to deploy multiple
+ceph-csi-operator that are configured to watch for the namespaces where they
+are deployed.
+
+## Development Methods
+
+Use operator framework which is higly used in operator developed and also by
+following Kubernetes best practices for operator development.
+
+## Architecture
+
+```mermaid
+graph TD
+    A[CRD Changes] --> B[Operator]
+    B -->  D[Configure CephFS CSI Driver]
+    B -->  E[Configure RBD CSI Driver]
+    B -->  F[Configure NFS CSI Driver]
+
+```
+
+In this updated diagram:
+
+- **CRD Changes**: Represents changes made to Custom Resource Definitions,
+  which trigger actions in the operator.
+- **Operator**: Listens for changes in CRDs and initiates the installation and
+  configuration of CSI drivers.
+- **Configure CephFS, NFS, RBD**: Actions performed by the operator to install
+  and configure the respective CSI drivers based on the CRD changes.
+
+## CRD's for ceph-csi-operator
+
+### CephCSIOperatorConfig CRD
+
+Manages operator-level configurations and default settings for CSI drivers.
+Ensures consistent default configurations across all CSI drivers.  This CRD is
+a namespace scoped CRD and a single CR instance should be created by the admin
+in the namespace where operator is running. The configurations are catagorized
+into 4 different parts
+
+- The operator configurations
+- Common configurations across all csi driver
+- Provisioner configurations
+- The Plugin Configurations
```

Comment:
These 2 are part of the common configuration, I am not sure why we need to differentiate them. 
```suggestion
```

Reply:
Yes thats correct, this is coved in common section